### ISSUE TITLE
running xmodmap to swap capslock and ctrl gets messy

### ISSUE THREAD

**Issue by bvk** [ch11ng/exwm#23]
I use the following xmodmap configuration to swap capslock and control keys:

remove Lock = Caps_Lock
remove Control = Control_L
keysym Control_L = Caps_Lock
keysym Caps_Lock = Control_L
add Lock = Caps_Lock
add Control = Control_L

When I attach an external keyboard and run xmodmap to swap the keys, it makes exwm unusable. I cannot switch to other window with C-x o

**Comment by ch11ng** [comment]
Did the code before @24b964b e.g. @048994c work?

**Comment by ch11ng** [comment]
And how about attaching the external keyboard before starting EXWM?

**Comment by bvk** [comment]
The older code at revision 048994c also didn't work either. exwm has same behavior.

My .emacs has the following code after (exwm-enable) that swaps capslock and control keys immediately after starting exwm which is working fine as expected.

...
(exwm-enable)
...
;; Swap control and capslock keys using xmodmap tool.
(defun wm-xmodmap()
  (call-process "xmodmap" nil (get-buffer-create "*wm*") nil
								(expand-file-name "~/config/xmodmap/swap-caps-with-ctrl")))
(wm-xmodmap)
...

I first noticed the issue when I disconnected and reconnected my keyboard. So, I commented out invoking (wm-xmodmap) was able to reproduce the same behavior using s-& and running "xmodmap ~/config/xmodmap/swap-caps-with-ctrl" command.
